### Step 1. 필요한 라이브러리 임포트

In [1]:
import pandas as pd

### Step 2. [데이터](https://raw.githubusercontent.com/justmarkham/DAT8/master/data/u.user) 확인

- user_id 유저 아이디
- age 나이
- gender 성별
- occupation 직업
- zip_code 우편번호

### Step 3. 

**read_csv를 통해 데이터를 올바른 형태로 불러온 뒤 users에 저장해주세요.**

user_id를 인덱스로 설정해주세요.

In [2]:
pd.read_csv?

In [56]:
url = 'https://raw.githubusercontent.com/justmarkham/DAT8/master/data/u.user'

users = pd.read_csv(url)
#users = pd.read_csv(url,sep='l', index_col='user_id')

### Step 4. 초반 25행 조회

In [5]:
users.head(25)

user_id|age|gender|occupation|zip_code
0                 1|24|M|technician|85711
1                      2|53|F|other|94043
2                     3|23|M|writer|32067
3                 4|24|M|technician|43537
4                      5|33|F|other|15213
5                  6|42|M|executive|98101
6              7|57|M|administrator|91344
7              8|36|M|administrator|05201
8                    9|29|M|student|01002
9                    10|53|M|lawyer|90703
10                    11|39|F|other|30329
11                    12|28|F|other|06405
12                 13|47|M|educator|29206
13                14|45|M|scientist|55106
14                 15|49|F|educator|97301
15            16|21|M|entertainment|10309
16               17|30|M|programmer|06355
17                    18|35|F|other|37212
18                19|40|M|librarian|02138
19                20|42|F|homemaker|95660
20                   21|26|M|writer|30068
21                   22|25|M|writer|40206
22                   23|30|F|artist|48197
23                   24|21|F|artist|94533
24                 25|39|M|engineer|55107

### Step 5. 마지막 10행 조회

In [10]:
users.tail(10)

user_id|age|gender|occupation|zip_code
933                934|61|M|engineer|22902
934                  935|42|M|doctor|66221
935                   936|24|M|other|32789
936                937|48|M|educator|98072
937              938|38|F|technician|55038
938                 939|26|F|student|33319
939           940|32|M|administrator|02215
940                 941|20|M|student|97229
941               942|48|F|librarian|78209
942                 943|22|M|student|77841

### Step 6. 데이터셋은 몇 개의 행으로 이루어져 있나요?

943

In [37]:
#shape()행과 열을 알수 있다.
users.shape[0]

943

In [16]:
len(users)

943

### Step 7.  데이터셋은 몇 개의 열로 이루어져 있나요?

4

In [38]:
#shape[]행과 열을 알수 있다.
users.shape[1]

1

### Step 8. 열(column)의 이름들을 모두 조회해주세요

Index(['age', 'gender', 'occupation', 'zip_code'], dtype='object')

In [39]:
users.columns

Index(['user_id|age|gender|occupation|zip_code'], dtype='object')

### Step 9. 데이터셋의 인덱스를 조회해주세요

Int64Index([  1,   2,   3,   4,   5,   6,   7,   8,   9,  10,
            ...
            934, 935, 936, 937, 938, 939, 940, 941, 942, 943],
           dtype='int64', name='user_id', length=943)

In [42]:
# "the index" (aka "the labels")   열 column 행index
users.index

RangeIndex(start=0, stop=943, step=1)

### Step 10. 각 열의 데이터타입은 무엇인가요?

In [44]:
#데이터 타입 (dtypes)을 사용 해라 
users.dtypes


user_id|age|gender|occupation|zip_code    object
dtype: object

### Step 11. occupation column만 조회해주세요

In [50]:
#조회하는 연산자 []
#users['occupation']
users.occupation

AttributeError: ignored

### Step 12. 데이터셋에는 몇 종류의 직업이 존재하나요?

21

In [60]:
users.occupation.unique()
#users.occupation.nunique()유니크에 앞에n을 쓰면 개수 까지 보여줌 유니크는 occupation의 맻 종류의 직업을 보여줌

AttributeError: ignored

### Step 13. 데이터셋에서 가장 많이 존재하는 직업은 무엇인가요?

student

In [64]:
#Because "most" is asked
#or
#to have the top 5
users.occupation.value_counts().head(1).index[0]


AttributeError: ignored

### Step 14. users.describe() 코드를 실행해서 숫자 열의 요약정보를 조회해주세요

In [65]:
#Notice: by default, only the numeric columns are returned. 
users.describe()

user_id|age|gender|occupation|zip_code
count                                     943
unique                                    943
top              799|49|F|administrator|19716
freq                                        1

### Step 15. users.describe(include = 'all') 코드를 실행해서 모든 열의 정보를 조회해주세요

In [70]:
#Notice: By default, only the numeric columns are returned.
users.describe(include = 'all')

user_id|age|gender|occupation|zip_code
count                                     943
unique                                    943
top              799|49|F|administrator|19716
freq                                        1

### Step 16. occupation 칼럼만 조회해서 describe() 함수를 실행해주세요

count         943<br>
unique         21<br>
top       student<br>
freq          196<br>
Name: occupation, dtype: object<br>

In [74]:
#users.describe(include = 'occupation)
users.occopation.describe()

AttributeError: ignored

### Step 17. user들의 평균 나이는 몇 살인가요?
34

In [77]:
users.age.meen()                              #rund함수는  소수점을 

AttributeError: ignored

### Step 18. 데이터셋에서 가장 적게 등장하는 나이는 몇 살인가요?

가장 적게 등장하는 나이는 1번씩 등장하는 나이이며  7, 66, 73, 10, 11살 입니다. 이러한 결과를 조회할수 있도록 데이터 조회하는 코드를 작성해주세요.

In [79]:
 #7, 10, 11, 66 and 73 years -> only 1 occurrence
 #users.occupation.value_counts().head(1).index[0]
 users.occupation.value_count(1)index[0]
 users.age.value_count(1)index[0]#head()tail()을 잘 활용하시오

SyntaxError: ignored